src/app/file-panel.js

```diff
@@ -108,14 +112,16 @@ function filepanel (appAPI, files) {
 
   var events = new EventManager()
   var element = template()
-  // TODO please do not add custom javascript objects, which have no
-  // relation to the DOM to DOM nodes
-  element.events = events
+
   fileExplorer.events.register('focus', function (path) {
     appAPI.switchToFile(path)
   })
 
-  return element
+  var api = { events: events }
+
+  container.appendChild(element)
+
+  return api
```

Comment:
oh yeah, you are right. that actually doesnt work anymore. interesting. i'll think about it and adapt the pull request. good catch, thx.